BlueBrain/Search
# Create minimal docs

Just an initial setup, can be populated/modified later. Motivated by [BBP guidelines](https://bbpteam.epfl.ch/project/spaces/pages/viewpage.action?pageId=45486102).


To test run the following

```bash
pip install -e .[dev]
cd docs
make clean && make html
```
There should be no sphinx warnings and the docs are going to lie under `docs/_build/html/index.html`

Additionally, with sphinx one can run doctests that were either defined in docstrings or in documentation rst files. See more [info](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html)

```bash
cd docs
make doctest
```
------
- [x] API directives generated (`docs/source/api/*`) - no need to do it again unless change in modules/subpackages
- [x] Setting up `docs/conf.py` with reasonable parameters
- [x] Setting up general structure `docs/index.rst`
- [x] Installation and docs generation instructions `docs/source/instructions.rst`
- [x] Minimal example illustrating doctests `docs/source/example.rst`
- [x] Run doctests on travis
- [x] Put things from readme to docs and only put how to generate docs in the readme
